src/plugin/main.c

```diff
@@ -366,15 +367,157 @@ static void browser_focus(void* vptr, bool focus)
+	case OBS_KEY_8: return '8';
+	case OBS_KEY_9: return '9';
+	case OBS_KEY_NUMASTERISK: return VK_MULTIPLY;
+	case OBS_KEY_NUMPLUS: return VK_ADD;
+	case OBS_KEY_NUMMINUS: return VK_SUBTRACT;
+	case OBS_KEY_NUMPERIOD: return VK_DECIMAL;
+	case OBS_KEY_NUMSLASH: return VK_DIVIDE;
+	case OBS_KEY_NUM0: return VK_NUMPAD0;
+	case OBS_KEY_NUM1: return VK_NUMPAD1;
+	case OBS_KEY_NUM2: return VK_NUMPAD2;
+	case OBS_KEY_NUM3: return VK_NUMPAD3;
+	case OBS_KEY_NUM4: return VK_NUMPAD4;
+	case OBS_KEY_NUM5: return VK_NUMPAD5;
+	case OBS_KEY_NUM6: return VK_NUMPAD6;
+	case OBS_KEY_NUM7: return VK_NUMPAD7;
+	case OBS_KEY_NUM8: return VK_NUMPAD8;
+	case OBS_KEY_NUM9: return VK_NUMPAD9;
+	case OBS_KEY_SEMICOLON: return VK_OEM_1;
+	case OBS_KEY_A: return 'A';
+	case OBS_KEY_B: return 'B';
+	case OBS_KEY_C: return 'C';
+	case OBS_KEY_D: return 'D';
+	case OBS_KEY_E: return 'E';
+	case OBS_KEY_F: return 'F';
+	case OBS_KEY_G: return 'G';
+	case OBS_KEY_H: return 'H';
+	case OBS_KEY_I: return 'I';
+	case OBS_KEY_J: return 'J';
+	case OBS_KEY_K: return 'K';
+	case OBS_KEY_L: return 'L';
+	case OBS_KEY_M: return 'M';
+	case OBS_KEY_N: return 'N';
+	case OBS_KEY_O: return 'O';
+	case OBS_KEY_P: return 'P';
+	case OBS_KEY_Q: return 'Q';
+	case OBS_KEY_R: return 'R';
+	case OBS_KEY_S: return 'S';
+	case OBS_KEY_T: return 'T';
+	case OBS_KEY_U: return 'U';
+	case OBS_KEY_V: return 'V';
+	case OBS_KEY_W: return 'W';
+	case OBS_KEY_X: return 'X';
+	case OBS_KEY_Y: return 'Y';
+	case OBS_KEY_Z: return 'Z';
+	case OBS_KEY_BRACKETLEFT: return VK_OEM_4;
+	case OBS_KEY_BACKSLASH: return VK_OEM_5;
+	case OBS_KEY_BRACKETRIGHT: return VK_OEM_6;
+	case OBS_KEY_ASCIITILDE: return VK_OEM_3;
+
+	case OBS_KEY_HENKAN: return VK_CONVERT;
+	case OBS_KEY_MUHENKAN: return VK_NONCONVERT;
+	case OBS_KEY_KANJI: return VK_KANJI;
+	case OBS_KEY_TOUROKU: return VK_OEM_FJ_TOUROKU;
+	case OBS_KEY_MASSYO: return VK_OEM_FJ_MASSHOU;
+
+	case OBS_KEY_HANGUL: return VK_HANGUL;
+
+	case OBS_KEY_BACKSLASH_RT102: return VK_OEM_102;
+
+	case OBS_KEY_MOUSE1: return VK_LBUTTON;
+	case OBS_KEY_MOUSE2: return VK_RBUTTON;
+	case OBS_KEY_MOUSE3: return VK_MBUTTON;
+	case OBS_KEY_MOUSE4: return VK_XBUTTON1;
+	case OBS_KEY_MOUSE5: return VK_XBUTTON2;
+
+	/* TODO: Implement keys for non-US keyboards */
+	default:;
+	}
+	return 0;
+}
+
 static void browser_key_click(void* vptr, const struct obs_key_event* event, bool key_up)
 {
 	struct browser_data* data = vptr;
 	char chr = 0;
 	if (event->text)
 		chr = event->text[0];
-	blog(LOG_INFO, "Key: %s %d %d", event->text, event->native_vkey, event->native_scancode);
+	blog(LOG_INFO, "Key: %s %d %d %d", event->text, event->native_vkey, event->native_scancode, custom_get_virtual_key(obs_key_from_virtual_key(event->native_vkey)));
 
-	browser_manager_send_key(data->manager, key_up, event->native_vkey, event->modifiers, chr);
+	browser_manager_send_key(data->manager, key_up, custom_get_virtual_key(obs_key_from_virtual_key(event->native_vkey)), event->modifiers, chr);
```

Comment:
It might be better to buffer the result of `custom_get_virtual_key()` to some variable as the function is quite big (performance).

Reply:
no need, since the value returned from that function is used only once